components/mpas-ocean/src/shared/mpas_ocn_gm.F

```diff
@@ -680,6 +681,38 @@ subroutine ocn_GM_compute_Bolus_velocity(statePool, &
          endif
 
          nEdges = nEdgesArray(3)
+         if(config_gm_enable_horizontal_resolution_function) then
+            call mpas_pool_get_array(meshPool, 'fEdge', fEdge)
+
+            !$omp parallel
+            !$omp do schedule(runtime) private(k, cell1, cell2, sumN2, lt1, lt2, c_mode1, resolution)
+            do iEdge = 1, nEdges
+               cell1 = cellsOnEdge(1, iEdge)
+               cell2 = cellsOnEdge(2, iEdge)
+               sumN2 = 0.0
+
+               do k = minLevelEdgeBot(iEdge)+1, maxLevelEdgeTop(iEdge)-1
+
+                  lt1 = 0.5_RKIND*(layerThickness(k,cell1) + layerThickness(k-1,cell1))
+                  lt2 = 0.5_RKIND*(layerThickness(k,cell2) + layerThickness(k-1,cell2))
+                  sumN2 = sumN2 + 0.5_RKIND*(lt1*sqrt(max(BruntVaisalaFreqTop(k,cell1),0.0_RKIND)) + &
+                                  lt2*sqrt(max(BruntVaisalaFreqTop(k,cell2),0.0_RKIND)))
+               end do
+
+               c_mode1 = max(1.0E-5_RKIND, sumN2/3.141592_RKIND)
+               cGMphaseSpeed(iEdge) = c_mode1
+               resolution = sqrt(c_mode1 / (fEdge(iEdge)**2.0 + betaEdge(iEdge)*c_mode1 + 1.0E-15_RKIND)) &
+                            / dcEdge(iEdge)
```

Comment:
@vanroekel Based on Hallberg (2013) `L = sqrt( c_g^2/ ( f^2 + 2*beta*c_g) )`, if `c_mode1 = c_g^2` then you need a 2 and a sqrt on the second c_mode1:
```
sqrt(c_mode1 / (fEdge(iEdge)**2.0 + 2*betaEdge(iEdge)*sqrt(c_mode1) + 1.0E-15_RKIND)
```

Reply:
Considering how complex this whole calculation is, I will save the Rossby Radius here as an output variable so we can check it.

Reply:
Maybe `c_mode1 = c_g` and not `c_mode1 = c_g^2`. Then line 721 would be missing a square, not a square root, and would be 
```
resolution = sqrt(c_mode1^2 / (fEdge(iEdge)**2.0 + 2*betaEdge(iEdge)*c_mode1 + 1.0E-15_RKIND)) / dcEdge(iEdge)
```

Reply:
That is what I was thinking too (that c_mode1=c_g).

Reply:
This is correct. `c_mode1=c_g`